plugins/powerd/fu-plugin-powerd.c

```diff
@@ -65,40 +65,39 @@ fu_plugin_destroy(FuPlugin *plugin)
 		g_object_unref(data->proxy);
 }
 
-static gboolean
-fu_plugin_powerd_refresh_cb(FuPlugin *plugin)
+static void
+fu_plugin_powerd_rescan(GVariant *parameters, FuPlugin *plugin)
 {
-	FuPluginData *data = fu_plugin_get_data(plugin);
 	FuContext *ctx = fu_plugin_get_context(plugin);
 	guint32 power_type;
 	guint32 current_state;
 	gdouble current_level;
-	g_autoptr(GVariant) powerd_response = NULL;
-
-	/* retrieving battery info from "GetBatteryState" method call to powerd */
-	powerd_response = g_dbus_proxy_call_sync(data->proxy,
-						 "GetBatteryState",
-						 NULL,
-						 G_DBUS_CALL_FLAGS_NONE,
-						 -1,
-						 NULL,
-						 G_SOURCE_REMOVE);
-	if (powerd_response == NULL) {
-		g_debug("battery information was not loaded");
-		return G_SOURCE_REMOVE;
+
+	if (parameters == NULL) {
+		g_debug("BatteryStatePoll signal has no contents");
```

Comment:
is this a programming error, or caused by an older version of powerd?

Reply:
It seems like a programming error, and it should be  `g_return_val_if_null` type of thing

Reply:
I think this is defensive programming and not actually expected to happen

Reply:
In that case - I would suggest `g_return_if_fail` for it